opencog/reasoning/RuleEngine/rule-engine-src/pln/ForwardChainer.cc

```diff
@@ -27,84 +27,127 @@
 
 #include <opencog/query/PatternMatch.h>
 #include <opencog/reasoning/RuleEngine/rule-engine-src/Rule.h>
+#include <opencog/guile/SchemeSmob.h>
 
 ForwardChainer::ForwardChainer(AtomSpace * as, string conf_path /*=""*/) :
-		as_(as) {
-	fcmem_ = new FCMemory(as_);
-	if (conf_path != "")
-		_conf_path = conf_path;
-	init();
+        as_(as)
+{
+    fcmem_ = new FCMemory(as_);
+    if (conf_path != "")
+        _conf_path = conf_path;
+    init();
 }
 
-ForwardChainer::~ForwardChainer() {
-	delete cpolicy_loader_;
-	delete fcmem_;
+ForwardChainer::~ForwardChainer()
+{
+    delete cpolicy_loader_;
+    delete fcmem_;
 }
 
-void ForwardChainer::init() {
-	cpolicy_loader_ = new JsonicControlPolicyParamLoader(as_, _conf_path);
-	cpolicy_loader_->load_config();
-	fcmem_->search_in_af_ = cpolicy_loader_->get_attention_alloc();
-	fcmem_->rules_ = cpolicy_loader_->get_rules();
-	fcmem_->cur_rule_ = nullptr;
+void ForwardChainer::init()
+{
+    cpolicy_loader_ = new JsonicControlPolicyParamLoader(as_, _conf_path);
+    cpolicy_loader_->load_config();
+    fcmem_->search_in_af_ = cpolicy_loader_->get_attention_alloc();
+    fcmem_->rules_ = cpolicy_loader_->get_rules();
+    fcmem_->cur_rule_ = nullptr;
+
+    // Provide a logger
+    log_ = NULL;
+    setLogger(new opencog::Logger("forward_chainer.log", Logger::FINE, true));
+}
```

Comment:
In case you don't know already (I think you do, but you wanted your own log file), you could log to "tmp/cogserver.log" using the default logger() (the single static instance).

```
logger().info("whatever");
```